pkg/controller/podautoscaler/replica_calculator.go

```diff
@@ -287,48 +287,87 @@ func (c *ReplicaCalculator) GetObjectMetricReplicas(currentReplicas int32, targe
 		// return the current replicas if the change would be too small
 		return currentReplicas, utilization, timestamp, nil
 	}
-	replicaCount = int32(math.Ceil(usageRatio * float64(currentReplicas)))
+
+	readyPodCount, err := c.getReadyPodsCount(namespace, selector)
+
+	if err != nil {
+		return 0, 0, time.Time{}, fmt.Errorf("unable to calculate ready pods: %s", err)
+	}
+
+	replicaCount = int32(math.Ceil(usageRatio * float64(readyPodCount)))
```

Comment:
With this implementation you can actually scale-down if your usageRatio is >1 and your readyPodCount < currentReplicas. This feels intuitively wrong - if the metric says we should scale-up we probably shouldn't be scaling down.

Reply:
this is interesting edge case (i believe it may appear due metric propagation delay), but still not clear how we can handle this. should we stay with existing replicas and allow only scale up if usageRatio > 1?

Reply:
Hmm good catch @MaciekPytel - to be honest, I'm not sure what the proper behavior is in this situation. To do nothing?

To me, it actually doesn't feel that troublesome that you would "scale down" - as I understand it, it would essentially be saying we need less pods that we predicted to handle this load (or the load has changed since then).

Happy to implement whatever you think is best :)

Reply:
@MaciekPytel @DirectXMan12 happy tuesday :) Just tagging to see what y'all think is the correct behavior in this circumstance?

Reply:
I thought about it some more and I didn't come up with any better solution that doesn't complicate things more than it's worth. Between the fact that the pods that haven't started yet shouldn't impact the metric and forbidden window protecting us from flapping number of replicas, I think this is actually good enough.

Sorry for the confusion.